### PR TITLE
[Accessibility] Replaced i with span tags

### PR DESCRIPTION
**Pull request by ghost**
This PR takes care of all the icon `<i>` tags in the Joomla core outputs, replacing them with `<span>` tags.
So `<i class="icon-..."></i>` becomes `<span class="icon-..."></span>`.

The `<i>` tags are not good from an accessibility point of view.
Replacing them with spans has no visual impact.

Testing the control panel via http://achecker.ca/checker:
Before: 

After:

To test, make sure all icons are displaying properly and that scripts that control icons still work ok.
Some scripts that have been tweaked to work with the spans:
- the update notification script (controlling the messages on the control panel left bottom) 
- the tree menu script controlling the folding icons in the menu item selection (in modules for example)

This can impact 3rd party extensions that might have javascript or css specifically written to expect `<i>` tags, like:

```
i.icon-home { ... }
```

Those would have to be changed to keep working.

### CODE REVIEW

**Comment by nonumber** [comment]
PS: This PR does not touch the FOF files.

**Comment by ihuibregtse** [comment]
Before testing I changed <span class="icon-lock hasTooltip" title="<?php echo JText::_('JGLOBAL_PASSWORD') ?>"></span> into <i class="icon-lock hasTooltip" title="<?php echo JText::_('JGLOBAL_PASSWORD') ?>"></i> in the default.php of the mod-login. The icon showed coreectly with the i-tag. After applying the patch, the i-tag was still there and wasn't replaced by a <span>-tag. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6467">issues.joomla.org/joomla-cms/6467</a>.</sub>

**Comment by nonumber** [comment]
Not sure what you mean. Why is test unsuccessful?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6467">issues.joomla.org/joomla-cms/6467</a>.</sub>

**Comment by ihuibregtse** [comment]
Sorry, my message seems to be broken. The patch didn't replace an i-tag I added to the login-form before testing.


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6467">issues.joomla.org/joomla-cms/6467</a>.</sub>

**Comment by nonumber** [comment]
That's not what this PR does. It fixes the <i> tags in Joomla's own output files. It doesn't dynamically replace you custom HTML.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6467">issues.joomla.org/joomla-cms/6467</a>.</sub>

**Comment by ihuibregtse** [comment]
Sorry! I'll check again later.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6467">issues.joomla.org/joomla-cms/6467</a>.</sub>

**Comment by spignataro** [comment]
I have checked this and it looks good. I can't find any issues. Can we get this moved to the master branch?

**Comment by wilsonge** [comment]
I like it :) Can we merge in staging please for the conflicts

**Comment by roland-d** [comment]
Staging is fine by me, however a second test is needed (or did you test @wilsonge ?). The PR needs to be synced please.

**Comment by nonumber** [comment]
Merge conflicts fixed.

**Comment by roland-d** [comment]
Just wanted to test this so we can move it forward but I get the following errors trying to apply the patch:
error: patch failed: administrator/components/com_categories/models/fields/modal/category.php:148
error: administrator/components/com_categories/models/fields/modal/category.php: patch does not apply
error: patch failed: administrator/components/com_contact/models/fields/modal/contact.php:143
error: administrator/components/com_contact/models/fields/modal/contact.php: patch does not apply
error: patch failed: administrator/components/com_content/models/fields/modal/article.php:138
error: administrator/components/com_content/models/fields/modal/article.php: patch does not apply
error: patch failed: administrator/components/com_newsfeeds/models/fields/modal/newsfeed.php:144
error: administrator/components/com_newsfeeds/models/fields/modal/newsfeed.php: patch does not apply
warning: administrator/components/com_redirect/views/links/tmpl/default.php has type 100644, expected 100755

Can you check please?

**Comment by nonumber** [comment]
Should be ok now.

**Comment by roland-d** [comment]
Thanks @nonumber I will test it tonight.

**Comment by roland-d** [comment]
@test all good. Can't find any issues. It's a good first step towards accessibility but a lot more to go :)

**Comment by dgt41** [comment]
@roland-d can you also take a look at https://github.com/joomla/joomla-cms/issues/6758 although it's an issue there is some code there (and it's working). That one has more impact in accessibility (not that this one is minor)

**Comment by n9iels** [comment]
@nonumber looks like this PR broke #6264 by accident. Can you look at this?

**Comment by roland-d** [comment]
@n9iels Are you sure that is the correct number, I don't see any reference to this PR in #6264. There was one issue with an undefined notice but that has been fixed.

File: administrator/components/com_users/helpers/html/users.php [hunk 1]
```diff
@@ -77,8 +77,8 @@ public static function filterNotes($count, $userId)
 
 		$title = JText::_('COM_USERS_FILTER_NOTES');
 
-		return '<a href="' . JRoute::_('index.php?option=com_users&view=notes&filter_search=uid:' . (int) $userId) . '" class="hasTooltip btn btn-mini" title="' . $title . '">'
-			. '<i class="icon-filter"></i></a>';
+		return '<a href="' . JRoute::_('index.php?option=com_users&view=notes&filter_search=uid:' . (int) $userId) . '">'
+			. JHtml::_('image', 'admin/filter_16.png', 'COM_USERS_NOTES', array('title' => $title), true) . '</a>';
```
Comment (n9iels) [inline administrator/components/com_users/helpers/html/users.php]: @roland-d I mean this line. It looks like this on my test version:
knipsel

In #6264 this icon was removed and replace with a font-icon

Comment (nonumber) [inline administrator/components/com_users/helpers/html/users.php]: Fix proposed with PR https://github.com/joomla/joomla-cms/pull/7059

**Pull request closed without merging**